django-stubs/contrib/admin/sites.pyi

```diff
@@ -39,28 +54,28 @@ class AdminSite:
     ) -> None: ...
     def unregister(self, model_or_iterable: Union[Type[Model], Iterable[Type[Model]]]) -> None: ...
     def is_registered(self, model: Type[Model]) -> bool: ...
-    def add_action(self, action: Callable, name: Optional[str] = ...) -> None: ...
+    def add_action(self, action: _ActionCallback, name: Optional[str] = ...) -> None: ...
     def disable_action(self, name: str) -> None: ...
     def get_action(self, name: str) -> Callable: ...
     @property
-    def actions(self): ...
+    def actions(self) -> Iterable[Tuple[str, _ActionCallback]]: ...
     @property
-    def empty_value_display(self): ...
+    def empty_value_display(self) -> str: ...
     @empty_value_display.setter
-    def empty_value_display(self, empty_value_display: Any) -> None: ...
+    def empty_value_display(self, empty_value_display: str) -> None: ...
     def has_permission(self, request: WSGIRequest) -> bool: ...
     def admin_view(self, view: Callable, cacheable: bool = ...) -> Callable: ...
     def get_urls(self) -> List[URLResolver]: ...
     @property
     def urls(self) -> Tuple[List[URLResolver], str, str]: ...
-    def each_context(self, request: Any): ...
+    def each_context(self, request: WSGIRequest) -> Dict[str, Any]: ...
```

Comment:
Are you sure it is not `HTTPRequest`?

Reply:
Other request arguments in this file are typed as `WSGIRequest`, so I picked that for consistency. I don't know when one should be preferred over the other.

Do you think I can change all of them to `HttpRequest`?

Reply:
I am really not sure. Let's keep it this way for now. Thanks!